packages/react-core/src/components/Toolbar/ToolbarItem.tsx

```diff
@@ -210,20 +232,48 @@ export const ToolbarItem: React.FunctionComponent<ToolbarItemProps> = ({
             isAllExpanded && styles.modifiers.expanded,
             isOverflowContainer && styles.modifiers.overflowContainer,
-            formatBreakpointMods(visibility, styles, '', getBreakpoint(width)),
-            formatBreakpointMods(align, styles, '', getBreakpoint(width)),
-            formatBreakpointMods(gap, styles, '', getBreakpoint(width)),
-            formatBreakpointMods(columnGap, styles, '', getBreakpoint(width)),
-            formatBreakpointMods(rowGap, styles, '', getBreakpoint(width)),
-            formatBreakpointMods(rowWrap, styles, '', getBreakpoint(width)),
             alignItems === 'start' && styles.modifiers.alignItemsStart,
             alignItems === 'center' && styles.modifiers.alignItemsCenter,
             alignItems === 'baseline' && styles.modifiers.alignItemsBaseline,
             alignSelf === 'start' && styles.modifiers.alignSelfStart,
             alignSelf === 'center' && styles.modifiers.alignSelfCenter,
             alignSelf === 'baseline' && styles.modifiers.alignSelfBaseline,
-            className
+            formatBreakpointMods(visibility, styles, '', getBreakpoint(width)),
+            formatBreakpointMods(align, styles, '', getBreakpoint(width)),
+            formatBreakpointMods(gap, styles, '', getBreakpoint(width)),
+            formatBreakpointMods(columnGap, styles, '', getBreakpoint(width)),
+            formatBreakpointMods(rowGap, styles, '', getBreakpoint(width)),
+            formatBreakpointMods(rowWrap, styles, '', getBreakpoint(width)),
+            className,
+            formatBreakpointMods(flexGrow, styles, '', getBreakpoint(width))
           )}
+          style={{
+            ...style,
+            ...(widths
+              ? setBreakpointCssVars(
+                  Object.entries(widths).reduce(
+                    (acc, [bp, size]) => {
+                      if (!size) {
+                        return acc;
+                      }
+                      const cssVarValueMap: Record<string, string> = {
+                        sm: 'var(--pf-c-toolbar__item--m-w-sm--Width)',
+                        md: 'var(--pf-c-toolbar__item--m-w-md--Width)',
+                        lg: 'var(--pf-c-toolbar__item--m-w-lg--Width)',
+                        xl: 'var(--pf-c-toolbar__item--m-w-xl--Width)',
+                        '2xl': 'var(--pf-c-toolbar__item--m-w-2xl--Width)',
+                        '3xl': 'var(--pf-c-toolbar__item--m-w-3xl--Width)',
+                        '4xl': 'var(--pf-c-toolbar__item--m-w-4xl--Width)'
+                      };
+                      const value = cssVarValueMap[size as keyof typeof cssVarValueMap];
+                      return value ? { ...acc, [bp]: value } : acc;
+                    },
+                    {} as Record<string, string>
+                  ),
+                  (c_toolbar__item_Width as any).name
+                )
+              : undefined)
+          }}
```

Comment:
Related to the above comment, we'll need to tweak this and we can simplify it a bit :

```suggestion
         style={{
            ...style,
            ...(widths ? setBreakpointCssVars(widths, c_toolbar__item_Width.name) : undefined)
          }}
```